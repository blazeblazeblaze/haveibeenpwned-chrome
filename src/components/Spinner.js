import React from 'react';
import PropTypes from 'prop-types';

const Spinner = ({src}) => {
  return(
    <div className='spinner-wrapper'>
      <img src={src} className="spinner" alt="spinner" />
    </div>
  )
};

Spinner.propTypes = {
  src: PropTypes.string.isRequired
};

Spinner.defaultProps = {
  src: 'src.png'
};

export default Spinner;
