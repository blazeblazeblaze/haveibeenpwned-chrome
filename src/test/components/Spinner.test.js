import React from 'react';
import ReactDOM from 'react-dom';
import Spinner from '../../components/Spinner';
import renderer from 'react-test-renderer';

test('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Spinner />, div);
  ReactDOM.unmountComponentAtNode(div);
});

test('Spinner uses default spinner file path', () => {
  const component = renderer.create(
   <Spinner />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test('Spinner uses custom spinner file path', () => {
  const component = renderer.create(
   <Spinner src='some-path.png' />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
