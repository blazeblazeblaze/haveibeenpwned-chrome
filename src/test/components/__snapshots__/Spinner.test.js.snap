// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Spinner uses custom spinner file path 1`] = `
<div
  className="spinner-wrapper"
>
  <img
    alt="spinner"
    className="spinner"
    src="some-path.png"
  />
</div>
`;

exports[`Spinner uses default spinner file path 1`] = `
<div
  className="spinner-wrapper"
>
  <img
    alt="spinner"
    className="spinner"
    src="src.png"
  />
</div>
`;
